### PR TITLE
Handle Units

### LINKED ISSUE #14: Set scale in import settings
When importing a model from Rhino that is made in millimeters it comes into Blender 1000 times larger than expected. I suppose it's because of Blender using meters as default. 
So an option to set the scale in the import options would be nice.

### PR DESCRIPTION
**Pull request by joel-putnam**
# Handle Units on Import

Add in three options:
- Default: This will import the Rhino file and scale the objects relative to the existing Blender file units system. 1 foot in Rhino will be 0.3048 meters in Blender.
- Convert Rhino: This will import the Rhino file and scale the objects assuming the current Blender file units system. 1 foot in Rhino will be 1 meter in Blender.
- Convert Blender: This will re scale the existing Blender file and change the units to the incoming Rhino file. 1 meter in Blender will now be 1 foot as well 1 foot in Rhino will be 1 foot in Blender. 

## fixes / resolves
Resolves  #14

### CODE REVIEW

**Comment by jesterKing** [comment]
I have created a PR @ mcneel/rhino3dm#64 to simplify the unit conversion code.  There shouldn't have to be the need for typing out all factors manually, then write code to do that between the different systems, when we already have code for that in OpenNURBS.

**Comment by jesterKing** [comment]
@joel-putnam The PR @ mcneel/rhino3dm#64 has been merged now. I'm assuming new PyPI packages will be up within a couple of days. We should be able to get your unit conversion code much cleaner and have less points of potential failure (much easier to just blame OpenNURBS ;) )

Anyway the new rhino3dm.UnitSystem.UnitScale(from, to) function gives you now a scale factor without having to type out manually all the different back-and-forth numbers. The function is quite elaborate and takes many, many cases into account, including the adaptive ones (not sure how exactly, but I know it does).

**Comment by joel-putnam** [comment]
Great. I am glad that is managed somewhere else. I will update the package and rework the code. I will get back to you when its ready for a review.

**Comment by joel-putnam** [comment]
So it looks like the pip package was not updated to reflect your pull request for rhino3dm ? Sorry due to the holiday it took me some time to get back to this.

**Comment by joel-putnam** [comment]
this has taken entirely too long but I have now switched over to using the new 
rhino3dm.UnitSystem.UnitScale(from, to) 
function

joel

**Comment by jesterKing** [comment]
Ok, there is some confusion on how the Blender unit settings work. You are using `unit_settings.length_unit` to determine how things should be scaled, but that setting (and `mass_unit`, `system`, `system_rotation` and `time_unit`) are purely for governing how data is shown in the GUI. Say you have Metric as the system, scale_factor 1.0 and the length_unit set to millimeter the underlying data is still in meter, but it is shown as mm. so a plane of 1.0 x 1.0 will show as 1000mm x 1000mm. The underlying edge lengths will still be 1.0 x 1.0.

In other words, the underlying data system is *always* in Blender Units, which essentially equals 1m x 1m. So conversions should always be made with that in mind.

**Comment by jesterKing** [comment]
This has been superseded by #24 that is now merged.

**Comment by jesterKing** [comment]
Many thanks for your efforts, I hope we will see more additions.

**Comment by tsvilans** [comment]
> Many thanks for your efforts, I hope we will see more additions.
🙌 🎆
